### DISCUSSION TITLE
[New Feature] backport the naval locomotor from YR to TS

### DISCUSSION THREAD

**Discussion by Paladin-Bloodmoon** [Vinifera-Developers/Vinifera#1318] [Feature Requests]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I have attached as much information as possible *(screenshots, debug and exception logs, etc)*.

### Description

what it says in the title, this is just something that should've been in TS in the beginning.

### Steps To Reproduce

literally try to make a naval unit

### Expected Behaviour

naval units should only pathfind through water

### Actual Behaviour

the naval unit pathfinding works as if it were an amphibious vehicle, it attempts to cross land

**Comment by ZivDero** [comment]
The naval locomotor's benefits are unclear as it's mostly a copy of Drive and has nothing to do with naval pathfinding.
